static fov feature

eatmyvenom opened the pull request:
A simple bit of code to allow for static fov in sodium.
No performance improvement is made by this and is only here as a feature

venomsOther committed:
add static fov mixin and option in sodium settings

venomsOther committed:
comments

TheGlitch76 commented on src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumGameOptions.java:
```diff
@@ -31,6 +31,7 @@
         public boolean enableVignette = true;
         public boolean enableFog = true;
         public boolean enableClouds = true;
+        public boolean staticFov = true;
```
This should be false by default since it's a cosmetic change to vanilla behavior, IMO.

eatmyvenom replied:
Yeah that's true, I'll change this over in a few.

venomsOther committed:
change static fov to off by default

jellysquid3 committed:
Prohibit using OptiFabric with Sodium

This is for your own protection, really. You don't want
to pick up the 15 acres of broken glass that will result
from trying to use them together.

jellysquid3 committed:
Do not upload empty chunk meshes if render data doesn't already exist

jellysquid3 committed:
Prevent chunk graph nodes from being created outside world bounds

jellysquid3 committed:
Flatten ChunkGraphNode type into ChunkRender

Reduces pointer chasing and memory allocations

jellysquid3 committed:
Refactor ChunkMeshInfo and improve code clarity

jellysquid3 committed:
Remove deprecated chunk status tracker

jellysquid3 committed:
Do not allocate GPU resources for empty mesh layers

jellysquid3 committed:
Drop references to GL objects immediately after deleting them

jellysquid3 committed:
Add patches to improve performance of Mojmath

jellysquid3 committed:
Store chunk render neighbors to avoid map lookups

jellysquid3 committed:
Use chunk load/unload events to track chunk render status

jellysquid3 committed:
Implement chunk unloading using events

jellysquid3 committed:
Use the correct coordinates when requesting chunk updates

jellysquid3 committed:
Fix chunks being improperly culled for being "too far away"

jellysquid3 committed:
Use improved block data container for rendering blocks

Do not schedule render jobs for empty chunks

jellysquid3 committed:
Initial work for throttling chunk uploads

jellysquid3 committed:
Implement basic block entity rendering

jellysquid3 committed:
Add patches to improve performance of MatrixStack operations

jellysquid3 committed:
Implement faster vertex write method for VertexBuffer

jellysquid3 committed:
Avoid retrieving chunk layer information multiple times every render call

jellysquid3 committed:
Add patches to further improve entity rendering hot path

jellysquid3 committed:
fix: Do not attempt to render chunks without render information

jellysquid3 committed:
fix: Encode quad normals in the correct order

jellysquid3 committed:
fix: Implement fallback path when BufferBuilder#vertex can't straight copy data

jellysquid3 committed:
new: Implement specialized Matrix rotation methods

jellysquid3 committed:
change: Allow usage of Unsafe to be disabled

jellysquid3 committed:
fix: Use workaround to see if a chunk is present on the client

jellysquid3 committed:
fix: Many fixes and improvements for chunk rendering

This super-commit is an accident of poor hygiene and laziness.
- Fixes chunk statuses failing to be updated after a render reload, leading
  to chunks not appearing correctly in the world
- Implement code path to remove meshes from a chunk data container, fixes
  blocks not disappearing when they are the last to be removed from a layer
- Fixes a number of scheduling issues and thread-safety problems present in
  the vanilla game
- Fixes problems where occlusion data would be grossly incorrect after a
  chunk update
- Move some data regarding chunks status to a new column data container to
  avoid an excessive amount of visitations in order to update chunk statuses

jellysquid3 committed:
new: Implement additional optimizations for matrix uploading

jellysquid3 committed:
new: Use XoShiRoRandom for client block ticking

jellysquid3 committed:
new: Avoid excessive frustum checks by storing visibility information for columns

jellysquid3 committed:
fix: Use more accurate method for selecting region bounds

Fixes chunks sometimes not fully rendering within the render distance

jellysquid3 committed:
new: Copy pointers to light arrays into ChunkSlice for rendering

venomsOther committed:
Merge branch '1.15.x' of git://github.com/jellysquid3/sodium into jellysquid3-1.15.x

venomsOther committed:
Merge branch 'jellysquid3-1.15.x'

venomsOther committed:
accidentally put in something that doesnt exist lul

jellysquid3 committed:
fix: Use correct coordinate values in ChunkSlice when retrieving light data

jellysquid3 committed:
style: Apply code style guidelines, add EditorConfig file

jellysquid3 committed:
change: Try to complete more work on the main thread while waiting on jobs to complete

jellysquid3 committed:
change: Use optimized YZX iteration order for block data

jellysquid3 committed:
fix: Notify the chunk graph that the render distance has changed

jellysquid3 committed:
fix: Unload chunk render data when the backing chunk is unloaded

jellysquid3 committed:
new: Implement our own chunk manager on the client

Should hopefully resolve issues with chunk statuses not being
sent over to the renderer correctly

jellysquid3 committed:
fix: Implement BiomeAccess support in ChunkSlice

Prevents the chunk renderer threads from trying to access state
on the main renderer thread as they no longer have to use a
fallback path which goes through the world

jellysquid3 committed:
fix: Allow passing a custom BiomeAccess to ClientWorld#getColor

This code is awful.

jellysquid3 committed:
fix: Correct issues with smooth lighting when blocks are adjacent to water

jellysquid3 committed:
new: Implement "tick only visible textures" feature

jellysquid3 committed:
change: Update to Gradle 6.3, use Java 14

jellysquid3 committed:
refactor: Cleanup and document some interfaces, remove unnecessary Mixins

jellysquid3 committed:
fix: Correctly handle dead-end cases in the chunk graph

jellysquid3 committed:
new: Implement optimized colorization cache for chunk meshing

jellysquid3 committed:
new: Include optimizations for particle rendering

jellysquid3 committed:
change: Cache most recently accessed chunk in the client chunk manager

jellysquid3 committed:
fix: Ensure the client's chunk manager updates the render distance

jellysquid3 committed:
fix: Implement more thorough fallback paths for direct vertex writing

jellysquid3 committed:
refactor: Force spaces for indentation absolutely everywhere

jellysquid3 committed:
fix: Modify biome caching implementation to allow neighboring chunks to be accessed

jellysquid3 committed:
fix: Use atomic operations in the biome cache

jellysquid3 committed:
change: Remove unnecessary lock in BiomeCacheManager

jellysquid3 committed:
change: Do not allow vanilla's renderer to allocate some resources

Prevents around 20MB of allocations and costly GL object initialization on reload

jellysquid3 committed:
new: Use arenas to reduce object allocation rate, reduce memory bandwidth spent on matrix operations

jellysquid3 committed:
fix: Use the correct camera setup logic for billboard particles

venomsOther committed:
????

venomsOther committed:
Merge branch '1.15.x' of https://github.com/jellysquid3/sodium-fabric

jellysquid3 committed:
new: Allow entity rendering to use Sodium's block renderer

jellysquid3 committed:
new: Allow different light cache implementations, use hashing for entity rendering

jellysquid3 committed:
refactor: Make semantics of Arena more clear, add documentation

jellysquid3 committed:
refactor: Initial sweep for re-organizing code

jellysquid3 committed:
refactor: Apply code style guidelines

jellysquid3 committed:
new: Implement global block entity rendering

jellysquid3 committed:
fix: Implement fallback block rendering path for damage overlay rendering

jellysquid3 committed:
fix: Allow for chunk renderer implementations to be changed in-game

jellysquid3 committed:
fix: Only allow for fog occlusion to be used on supporting hardware

jellysquid3 committed:
new: Include patches for smooth entity lighting

jellysquid3 committed:
new: Add advanced entity culling patches

jellysquid3 committed:
fix: Use the correct BufferBuilderStorage to block damage rendering on tile entities

jellysquid3 committed:
fix: Use the correct visibility frame index for entity culling

venomsOther committed:
Merge branch '1.15.x' of https://github.com/jellysquid3/sodium-fabric

venomsOther committed:
I did it, it works

eatmyvenom committed:
Merge branch 'no-hurt-cam' into master
